megaparse/parser/unstructured_parser.py

```diff
@@ -0,0 +1,125 @@
+import asyncio
+from unstructured.partition.auto import partition
+from dotenv import load_dotenv
+from megaparse.parser import MegaParser
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_core.prompts import ChatPromptTemplate
+import re
+from langchain_openai import ChatOpenAI
+import os
+
+from megaparse.parser.type import StrategyEnum
+
+
+class UnstructuredParser(MegaParser):
+    load_dotenv()
+
+    def __init__(self, strategy=StrategyEnum.AUTO, model: BaseChatModel | None = None):
+        self.strategy = strategy
+        self.model = model
+
+    # Function to convert element category to markdown format
+    def convert_to_markdown(self, elements):
+        markdown_content = ""
+
+        for el in elements:
+            markdown_content += self.get_markdown_line(el)
+
+        return markdown_content
+
+    def get_markdown_line(self, el: dict):
+        element_type = el["type"]
+        text = el["text"]
+        metadata = el["metadata"]
+        parent_id = metadata.get("parent_id", None)
+        category_depth = metadata.get("category_depth", 0)
+        table_stack = []
+
+        if "emphasized_text_contents" in metadata:
+            print(metadata["emphasized_text_contents"])
+
+        # Markdown line defaults to empty
+        markdown_line = ""
+
+        # Element type-specific markdown content
+        markdown_types = {
```

Comment:
Define enum of markdown types

Reply:
Not sure to understand, why isn't the dict better ?

Reply:
You mean smth like that ? : 
class MarkDownType(str, Enum):
    """Markdown type enumeration."""

    TITLE = "Title"
    SUBTITLE = "Subtitle"
    HEADER = "Header"
    FOOTER = "Footer"
    NARRATIVE_TEXT = "NarrativeText"
    LIST_ITEM = "ListItem"
    TABLE = "Table"
    PAGE_BREAK = "PageBreak"
    IMAGE = "Image"
    FORMULA = "Formula"
    FIGURE_CAPTION = "FigureCaption"
    ADDRESS = "Address"
    EMAIL_ADDRESS = "EmailAddress"
    CODE_SNIPPET = "CodeSnippet"
    PAGE_NUMBER = "PageNumber"
    DEFAULT = "Default"
    UNDEFINED = "Undefined"

and then a dict ? Just to make sure there are no new types ?

Reply:
to see later, adds complexity (create a ticket)